### PR TITLE
Researchers

### PR DESCRIPTION
**Pull request by LetsTrie**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,2 +1,5 @@
 MONGODB_HOST = mongodb+srv://agriplus12345:agriplusxxx@cluster0-jn5sz.mongodb.net/feedback?retryWrites=true&w=majority
 JWT_SECRET = 123
+EMAIL_ID = sakibkhan111296@gmail.com
```
Comment (anisul-islam-prog) [inline .env]: don't put your email, password in `.env` file

File: config/config.js [hunk 2]
```diff
@@ -4,22 +4,26 @@ const dotenv = require("dotenv");
...
 
 const envSchema = Joi.object({
-	NODE_ENV: Joi.string()
-		.allow(["development", "production", "test", "provision"])
-		.default("development"),
-	PORT: Joi.number().default(3000),
-	MONGODB_HOST: Joi.string().required(),
-	JWT_SECRET: Joi.string().required()
+  NODE_ENV: Joi.string()
+    .allow(["development", "production", "test", "provision"])
```
Comment (anisul-islam-prog) [inline config/config.js]: What do you mean by provision?

File: config/sendingEmail.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+module.exports.sendEmailForResetPassword = async (username, emailID, token) => {
+  const link = `${host}/auth/forgetPassword/${token}`;
+  const mailOptions = {
+    from: config.emailID,
+    to: emailID,
+    subject: "Generated Token For Reset Password",
+    html: `
+      <h1> Hi ${username}, </h1>
+      <h2> Copy This Link and paste it into POSTMAN </h2>
+      <a href="${link}"> ${link} </a>
```
Comment (anisul-islam-prog) [inline config/sendingEmail.js]: Don't expose the api link here. Instead, send a serving page.

File: config/sendingEmail.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+module.exports.sendEmailForVerifyAccount = async (username, emailID, token) => {
+  const link = `${host}/auth/verifyAccount/${token}`;
+  const mailOptions = {
+    from: config.emailID,
+    to: emailID,
+    subject: "Generated Token For Account Verification",
+    html: `
+      <h1> Hi ${username}, </h1>
+      <h2> Copy This Link and paste it into POSTMAN </h2>
+      <a href="${link}"> ${link} </a>
```
Comment (anisul-islam-prog) [inline config/sendingEmail.js]: Same as previous.

File: index.routes.js [hunk 5]
```diff
@@ -2,8 +2,11 @@ const express = require("express");
 const accountRoutes = require("./server/account/account.route");
 const adminRoutes = require("./server/admin/admin.route");
 const researcherRoutes = require("./server/researcher/researcher.route");
+const audienceRoutes = require('./server/audience/audience.route');
```
Comment (anisul-islam-prog) [inline index.routes.js]: Follow the alphabetical orders [a-z] in routes level.

File: server/account/account.controller.js [hunk 6]
```diff
@@ -1,11 +1,11 @@
-const bcrypt = require("bcrypt");
+const bcrypt = require("bcryptjs");
```
Comment (anisul-islam-prog) [inline server/account/account.controller.js]: Use `bcrypt` module instead of `bcryptjs`.

File: server/account/account.controller.js [hunk 7]
```diff
@@ -17,84 +17,91 @@ const {
...
+        message: 'Choose designation between "admin", "researcher" & "audience"'
+      });
+    }
+    let user = await AccountModel.findOne({ email: data.email });
+    if (user) return res.status(400).json({ message: "Email already Exists" });
+    console.log(data, user);
+    user = await AccountModel.findOne({ phoneNumber: data.phoneNumber });
+    if (user)
+      return res.status(400).json({ message: "Phone Number already Exists" });
+    console.log(data, user);
```
Comment (anisul-islam-prog) [inline server/account/account.controller.js]: Don't use `console.log`

File: server/account/account.controller.js [hunk 8]
```diff
@@ -17,84 +17,91 @@ const {
...
+    const hashedPassword = await bcrypt.hash(data.password, 10);
+    const newUser = new AccountModel({
+      name: data.name,
+      email: data.email,
+      phoneNumber: data.phoneNumber,
+      role: data.role,
+      password: hashedPassword
+    });
+    await newUser.save();
+    newUser.password = undefined;
```
Comment (anisul-islam-prog) [inline server/account/account.controller.js]: Very Good thing @LetsTrie

File: .env (hunk 1 above)
Comment (LetsTrie) [reply .env]: done

File: config/config.js (hunk 2 above)
Comment (LetsTrie) [reply config/config.js]: orvi vai added this... i know the first three terms.

Comment (anisul-islam-prog) [reply config/config.js]: provision delete kore dao

File: server/account/account.controller.js (hunk 8 above)
Comment (LetsTrie) [reply server/account/account.controller.js]: It will undefined when it is returning the object.

File: server/account/account.controller.js (hunk 7 above)
Comment (LetsTrie) [reply server/account/account.controller.js]: ok

File: server/account/account.controller.js (hunk 6 above)
Comment (LetsTrie) [reply server/account/account.controller.js]: sudo npm i bcrypt => shows error.

**Pull request merged**